packages/wallet/src/Wallet.d.ts

```diff
@@ -0,0 +1,308 @@
+    >;
+    preferencesController: PreferencesController;
+    transactionController: TransactionController;
+  };
+
+  #services: {
+    // TODO: Create EVM RPC request service, extracting from fetch middleware
+    evmRpcRequest: () => void;
+    // TODO: Create Etherscan service, extracting from transaction controller
+  };
+
+  /**
+   * Construct a MetaMask wallet.
+   *
+   * @param options - Options.
+   * @param options.controllerMessenger - An unrestricted global messenger, used as the primary
+   * message broker for the wallet.
+   * @param options.state - The initial wallet state, broken down by controller.
+   */
+  constructor({
+    controllerMessenger,
+    state = {},
+  }: {
+    controllerMessenger: ControllerMessenger<AllWalletActions, AllWalletEvents>;
+    state?: Partial<
+      MetamaskState<
+        ControllerPermissionSpecification,
+        ControllerCaveatSpecification
+      >
+    >;
+  });
+
+  /**
+   * Initialize the wallet.
+   *
+   * This step is for asynchronous operations that should be performed after the wallet is
+   * initially constructed. For example, status checks, remote configuration updates, or preemptive
+   * caching.
+   *
+   * Note that this initialziation may not occur right away after wallet construction. For new
+   * wallet installations, this initialization will not be run until after onboarding.
+   */
+  initialize(): Promise<void>;
+
+  // TODO: Consider adding state reset to base controller
+
+  /**
+   * Reset all transient wallet state.
+   *
+   * This method is meant for applications that expect to automatically restart during typical
+   * operation (e.g. a wallet running in a service worker). Such applications can persist all
+   * wallet state, including transient state, to ensure caches are not cleared during these routine
+   * restarts. But after a fresh application start, we still want to have the ability to clear
+   * transient data that is not intended to be persisted.
+   *
+   * This method will erase all transient wallet state, leaving only persistent state. It should be
+   * called before initialization during a new application session.
+   */
+  resetState(): void;
+
+  /**
+   * Create a "provider engine" for the given subject.
+   *
+   * A provider engine is a JSON-RPC request handler for provider JSON-RPC requests. It can be
+   * used to construct a provider, or to implement a wallet API.
+   *
+   * @param options - Options
+   * @param options.origin - The origin of the subject.
+   * @param options.subjectType - The type of the subject.
+   */
+  createProviderEngine({
+    origin,
+    subjectType,
+  }: {
+    origin: string;
+    subjectType: SubjectType;
+  }): JsonRpcEngine;
+
+  // TODO: Add start/resume, pause, stop methods to control all polling/services
+
+  // Additional jotnotes:
+  //
+  // Step0: Rename controller messenger, write ADR about services and selectors, refactor guts of ComposableController into compose function
```

Comment:
>efactor guts of ComposableController into compose function

I haven't yet demonstrated in this PR how we'd combine the state of this wallet library with additional controllers. But my idea was to extract the internals of `ComposableController` into a function, which could be used both internally in the wallet library, and externally by the client to compose additional controller states.